Automattic/wp-calypso
# Refactor G Suite Dialog to use New User List Component

#### Changes proposed in this Pull Request

* Rename generic Component to `GoogleApps` to `GSuiteUpgrade`
* Completely refactor `GSuiteDialog` as a functional component using the new `GSuiteNewUserList` shared component
* Refactor parents ( `GSuiteUpgrade`, `GSuiteNudge` ) to use new props


#### Testing instructions

1. Open console, run `localStorage.setItem('debug', 'calypso:analytics:*');`, keep console open
2. Add a new site for your test account with a domain and plan, pause before checkout.
3. Purchase site and domain.
4. Confirm you are taken to `/checkout:siteSlug/with-gsuite/:domain/:reciept_id`
5. Note your receipt id
6. Confirm new User List component works, add a couple users and click Add Email/Continue
7. Confirm that all users were added to the cart correctly
8. Using React Dev Tools inspect the `CartItems` Element
9. Inspect the `cart` property, drilling down to `cart.products[0].extra`
10. Confirm the `extra` object has a `receipt_for_domain` property with the same receipt id you notes in step 5.
11. Finish checkout
12. Go to `/domains/add/:siteSlug` for any site with a paid plan
13. Chose a domain ( we will not be purchasing it )
14. Confirm you have been taking to `/domains/add/:domain/google-apps/:siteSlug`
15. Open console for event tracking
16. Click "Skip" and confirm `calypso_checkout_gsuite_upgrade_skip_button_click` event
17. Return to step 14.
18. Add some users and click "Yes, Add Email"
19. Confirm `calypso_checkout_gsuite_upgrade_add_email_button_click` event
20.  Confirm users in cart are correct
